### PR TITLE
Bug 1188544 - [Dialer][NGA] Remove navigation based on hashchange r=arcturus, gsvelto

### PR DESCRIPTION
**Pull request by borjasalguero**

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7e1337ceeb2351f611c82a6b9458b32fb42d30af

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2c5680ab1c513fdb335b409694d52f2cb5352635

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2c5680ab1c513fdb335b409694d52f2cb5352635

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e24231e006b5030d6aa302cbdd6cb9b34a6168f8

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8dfec020d326967433e8ec4129b6b8460de83353

File: apps/communications/dialer/js/navigation.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    get currentView() {
+      return currentView ? currentView.id : null;
+    },
+    go: function(destination) {
+      var view =
+        document.querySelector(
+          'div[aria-labelledby="' + destination + '"]'
+        );
+      if (!view) {
+        console.log('Abortamos mision xq "view" no está.');
```
Comment (arcturus) [inline apps/communications/dialer/js/navigation.js]: ;)

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7da3eedfaf8650d6fc92f9b4a2fd22702e616921

File: apps/communications/dialer/js/navigation.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  exports.Navigation =  {
+    get currentView() {
+      return currentView ? currentView.id : null;
+    },
+    go: function(destination) {
+      var view =
+        document.querySelector(
+          'div[aria-labelledby="' + destination + '"]'
+        );
```
Comment (gabrielesvelto) [inline apps/communications/dialer/js/navigation.js]: nit: I'm not super-fond of querying via the aria-labelledby attribute, besides if the view is not found we return silently without throwing an error or anything. I'd rather have a list of the possible destinations hardcoded into the Navigation object with clear short names (keypad, recents, contacts) which would then be mapped to the id. If ``destination`` is not among the list then we throw an error. E.g., from the outside:
```javascript
Navigation.show('keypad');
```
And internally:
```javascript
go:function(destination) {
  var view;
  switch (destination) {
    case 'keypad': view = document.getElementyById('keypad-panel'); break;
    // ...
    default: throw new Error("Navigating to an invalid destination: " + destination);
  }
}

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2f2f1dc42f3b881af5d371c062e6400ae7497d12

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cc672af092c30b3de43663c191fdd67b34b2bbc7

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5cffe28e55627509974fe94473fbfa2bf2ce5e8b

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8e085c10772dda136c07cc408cd9ad5c29a4d2e8

**Pull request merged**